### ISSUE TITLE
DeprecationWarning re full filename and extension

### ISSUE THREAD

**Issue by drmrbrewer** [felixge/node-dateformat#175]
I'm getting the following warning when running a node app with the latest version of `dateformat`:

	(node:20) [DEP0151] DeprecationWarning: Package /usr/src/app/node_modules/dateformat/ has a "main" field set to "lib/dateformat", excluding the full filename and extension to the resolved file at "lib/dateformat.js", imported from /usr/src/app/myapp.js.
	 Automatic extension resolution of the "main" field is deprecated for ES modules.

**Comment by rlj1202** [comment]
It will be solved by changing the following code.

package.json
```
...
  "license": "MIT",
  "type": "module",
  "main": "lib/dateformat.js",
...
```

Just specify the full file name and extension as the warning message says.

**Comment by drmrbrewer** [comment]
@rlj1202 indeed, it should be a simple fix (@chase-manning).  I believe that the change should go a little further than you suggest, by using the full relative path name including prefix: `"main": "./lib/dateformat.js"`.

Also, I think that `main` should be replaced by `exports` (see e.g. [here](
https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c#how-can-i-move-my-commonjs-project-to-esm)) but I don't understand enough about this to know if there are any unintended consequences of doing this (see e.g. [here](https://nodejs.org/api/packages.html#package-entry-points)).
